lib/ansible/module_utils/net_tools/nios/api.py

```diff
@@ -227,7 +227,7 @@ def run(self, ib_obj_type, ib_spec):
         if ib_obj_ref:
             if len(ib_obj_ref) > 1:
                 for each in ib_obj_ref:
-                    if each['ipv4addr'] == proposed_object['ipv4addr']:
+                    if ('ipv4addr' in each and proposed_object) and each['ipv4addr'] == proposed_object['ipv4addr']:
```

Comment:
This will not do the right thing. Probably what you want is something along the lines of `if ('ipv4addr' in each) and ('ipv4addr' in proposed_object) and each['ipv4addr'] == proposed_object['ipv4addr']:`

Reply:
Ah, got the issue my bad. Fixed

Reply:
@Qalthos can you please approve the PR have made the requested change.

Reply:
@Qalthos can you please review and approve.